File: backend/app/schemas/event.py
import uuid
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field

EventState = Literal["START", "END"]
EventLocation = Literal["INSIDE", "OUTSIDE", "NOT_APPLICABLE"]
EntryMethod = Literal["QUICK_LOG", "MANUAL"]


class EventCreate(BaseModel):
    dog_id: uuid.UUID
    event_type_id: int
    event_time: Optional[datetime] = None
    state: Optional[EventState] = None
    location: EventLocation = "NOT_APPLICABLE"
    treat_type_id: Optional[uuid.UUID] = None
    notes: Optional[str] = Field(default=None, max_length=500)
    entry_method: EntryMethod = "QUICK_LOG"


class EventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    dog_id: uuid.UUID
    event_type_id: int
    event_time: datetime
    state: Optional[str]
    location: str
    treat_type_id: Optional[uuid.UUID]
    notes: Optional[str]
    entry_method: str
    created_at: datetime

File: frontend/src/types/event.ts
export interface EventType {
  id: number;
  code: string;
  display_name: string;
  category: string;
  supports_state: boolean;
  supports_location: boolean;
  supports_treat: boolean;
  option_category: string | null;
  option_required: boolean;
  supports_numeric: boolean;
  numeric_label: string | null;
  numeric_required: boolean;
  allowed_units: string[];
  supports_severity: boolean;
  severity_required: boolean;
  start_label: string | null;
  end_label: string | null;
  default_enabled: boolean;
}

export interface SavedOption {
  id: string;
  dog_id: string;
  category: string;
  name: string;
  is_active: boolean;
}

export interface EventPreference {
  event_type: EventType;
  is_enabled: boolean;
  display_order: number | null;
}

export type EventState = "START" | "END";
export type EventLocation = "INSIDE" | "OUTSIDE" | "NOT_APPLICABLE";

export interface EventCreate {
  dog_id: string;
  event_type_id: number;
  event_time?: string;
  state: EventState | null;
  location: EventLocation;
  option_id: string | null;
  numeric_value: number | null;
  unit: string | null;
  severity: number | null;
  notes: string | null;
  entry_method: "QUICK_LOG" | "MANUAL";
}

export interface EventUpdate {
  event_time?: string;
  state?: EventState | null;
  location?: EventLocation;
  option_id?: string | null;
  numeric_value?: number | null;
  unit?: string | null;
  severity?: number | null;
  notes?: string | null;
}

export interface LoggedEvent {
  id: string;
  dog_id: string;
  event_type_id: number;
  event_type_code: string;
  event_type_name: string;
  event_time: string;
  state: EventState | null;
  location: EventLocation;
  option_id: string | null;
  option_name: string | null;
  numeric_value: string | null;
  unit: string | null;
  severity: number | null;
  notes: string | null;
  entry_method: "QUICK_LOG" | "MANUAL";
  created_at: string;
}
